tests/btb/recommendation/test_recommender.py

```diff
@@ -37,7 +37,11 @@ def test___init__(self, nmf_mock):
         np.testing.assert_array_equal(recommender.dpp_matrix, self.dpp_matrix)
         np.testing.assert_array_equal(recommender.dpp_ranked, self.dpp_ranked)
         assert recommender.n_components == self.n_components
-        assert recommender.matching_dataset is None
+        # matching dataset starts out with a random choice
+        assert recommender.matching_dataset is not None
+        assert(
```

Comment:
This should be covered by mocking `np.random` and forcing `randint` to return a specific value.

Then, during the assert phase, check only if the `matching_dataset` equals the corresponding dataset in `dpp_vector`.

Otherwise, the test does not truly check if the asigned dataset is the one that we wanted.
For example, if the index was hardcoded as in `self.matching_dataset = self.dpp_matrix[0, :]` the test would pass anyway.